src/vscode-bicep/src/utils/AzExtOutputChannel.ts

```diff
@@ -0,0 +1,93 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+import { IAzExtOutputChannel } from "@microsoft/vscode-azext-utils";
+import {
+  OutputChannel,
+  ViewColumn,
+  window,
+  workspace,
+  WorkspaceConfiguration,
+} from "vscode";
+import { removePropertiesWithPossibleUserInfoInDeployParams } from "./removePropertiesWithPossibleUserInfo";
+
+// https://github.com/microsoft/vscode-azuretools/blob/main/utils/src/AzExtOutputChannel.ts
+// with support to remove properties with possible user info before appendLine(..) is invoked on output channel.
+// TODO: revisit this when https://github.com/Azure/azure-sdk-for-net/issues/27263 is resolved.
+export function createAzExtOutputChannel(
+  name: string,
+  extensionPrefix: string
+): IAzExtOutputChannel {
+  return new AzExtOutputChannel(name, extensionPrefix);
+}
+
+class AzExtOutputChannel implements IAzExtOutputChannel {
+  public readonly name: string;
+  public readonly extensionPrefix: string;
+  private _outputChannel: OutputChannel;
+
+  constructor(name: string, extensionPrefix: string) {
+    this.name = name;
+    this.extensionPrefix = extensionPrefix;
+    this._outputChannel = window.createOutputChannel(this.name);
+  }
+
+  public replace(value: string): void {
+    this._outputChannel.replace(value);
+  }
+
+  public append(value: string): void {
+    this._outputChannel.append(value);
+  }
+
+  public appendLine(value: string): void {
+    const updatedValue =
+      removePropertiesWithPossibleUserInfoInDeployParams(value);
```

Comment:
nit: Not thrilled about this implementation, but it's good enough for now.  And at least it's clear what it's doing.